### PR TITLE
Prudhvi.tls fix

### PR DESCRIPTION
**Pull request by prudhvi**
Handling the case correctly only if CA for TLS is given but not cert and key.
Logging error if TLS can't be enabled successfully.

### CODE REVIEW

File: test/mycnf/full-dsn [hunk 1]
```diff
@@ -3,6 +3,5 @@ user     = U
 password = P
 host     = H
 port     = 33560
-ssl-ca   = CA
-ssl-cert = CERT
-ssl-key  = KEY
+# Path is relative to the test file factory_test.go
+ssl-ca   = ../test/mycnf/rds-master.pem
```
Comment (prudhvi) [inline test/mycnf/full-dsn:7]: Would be nice to get this to just `./rds-master.pem` but currently the path should be relative to the script that's using this full-dsn file.

File: dbconn/factory.go [hunk 2]
```diff
@@ -111,7 +111,13 @@ func (f factory) Make(cfg blip.ConfigMonitor) (*sql.DB, string, error) {
 
 	// Load and register TLS, if any
 	tlsConfig, err := cfg.TLS.LoadTLS()
-	if tlsConfig != nil && err != nil {
+
+	if err != nil {
+		blip.Debug("Error loading TLS Config for %s", cfg.MonitorId)
```
Comment (daniel-nichter) [inline dbconn/factory.go:116]: Classic error message mistake: an error that doesn't say what the error is :-)  Either add `err` to the debug, or if the caller debug's the error line, then this Debug is superfluous

**Review by daniel-nichter — APPROVED** [review]

File: dbconn/factory.go (hunk 2 above)
Comment (prudhvi) [reply dbconn/factory.go]: Yeah thought about it,  i will be printing the error twice, so didn't print it here , caller will be logging the returned error anyway.

Comment (prudhvi) [reply dbconn/factory.go]: Cleaned up by removing the whole debug, also I was looking for patterns to say logger.Error or logger.Warn but I only see them available in few file(main.go or blip.go) .Is that intentional? Most other files only have blip.Debug available.

**Pull request merged**